pkg/apis/ceph.rook.io/v1/scc.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2021 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"bytes"
+	_ "embed"
+
+	secv1 "github.com/openshift/api/security/v1"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/util/yaml"
+)
+
+const (
+	// Determines how far into the stream the decoder will look to figure out whether
+	// this is a JSON stream or YAML
+	walkThroughBufferSize = 10
+)
+
+var (
+	//go:embed template/scc.yaml
+	securityContextConstraintsFilePath []byte
+	//go:embed template/scc-csi.yaml
+	securityContextConstraintsCSIFilePath []byte
+)
+
+// NewSecurityContextConstraints returns a new SecurityContextConstraints for Rook-Ceph to run on
+// OpenShift.
+func NewSecurityContextConstraints() (*secv1.SecurityContextConstraints, error) {
+	var scc secv1.SecurityContextConstraints
+	err := yaml.NewYAMLOrJSONDecoder(bytes.NewReader(securityContextConstraintsFilePath), walkThroughBufferSize).Decode(&scc)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to unmarshal security context constraints template")
+	}
+
+	return &scc, nil
+}
+
+// NewCSISecurityContextConstraints returns a new SecurityContextConstraints for Ceph-CSI to run on
+// OpenShift.
+func NewCSISecurityContextConstraints() (*secv1.SecurityContextConstraints, error) {
```

Comment:
can we move this to the csi pkg/operator/ceph/csi folder for easy maintenance of CSI bits? and also move the template to csi folder